server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -328,8 +329,19 @@ private static Symbol removeParts(Map<Set<RelationName>, Symbol> queryParts, Rel
     }
 
     @Nullable
-    private static <V> V removeMatch(Map<Set<RelationName>, V> valuesByNames, Set<RelationName> names, RelationName nextName) {
+    private static <V> V removeMatch(Map<Set<AnalyzedRelation>, V> valuesByNames, Set<RelationName> names, RelationName nextName) {
+        // JoinNames here is a set with exactly 2 RelationNames: 2 next tables from the "optimally ordered list"
+//        See code in JoinOperators:
+//        final RelationName lhsName = it.next();
+//        final RelationName rhsName = it.next();
+//        Set<RelationName> joinNames = new HashSet<>();
+//        joinNames.add(lhsName);
+//        joinNames.add(rhsName);
         for (RelationName name : names) {
+            // valuesByNames key used to be a single RelationName from the joinPair left/right sides.
+            // we cannot use AnalyzedRelation here as a map key.
+            // to reflect old logic we should check that left/right are DocTableRelations and they aggreagted outpus is (Set.of(name, nextName))
```

Comment:
this is not the case as the only more or less fitting to the `Join` node `AnalyzedRelation` is `QueriedSelectRelation`